diff --git a/build/compiler.w b/build/compiler.w
--- a/build/compiler.w
+++ b/build/compiler.w
@@ -422,20 +422,8 @@ fn run_compiler_metadata_action(ctx: ActionCtx) -> i32:
         rsp = rsp ++ "-lm\n"
         rsp = rsp ++ "-static-libstdc++\n"
         rsp = rsp ++ "-static-libgcc\n"
-        rsp = rsp ++ "-lz\n"
-        rsp = rsp ++ "-lzstd\n"
-        rsp = rsp ++ "-lxml2\n"
-        ld_rsp = ld_rsp ++ "-Bstatic\n"
-        ld_rsp = ld_rsp ++ "-lstdc++\n"
-        ld_rsp = ld_rsp ++ "-lgcc\n"
-        ld_rsp = ld_rsp ++ "-lgcc_eh\n"
-        ld_rsp = ld_rsp ++ "-Bdynamic\n"
-        ld_rsp = ld_rsp ++ "-lpthread\n"
-        ld_rsp = ld_rsp ++ "-ldl\n"
-        ld_rsp = ld_rsp ++ "-lm\n"
-        ld_rsp = ld_rsp ++ "-lz\n"
-        ld_rsp = ld_rsp ++ "-lzstd\n"
-        ld_rsp = ld_rsp ++ "-lxml2\n"
+        rsp = rsp ++ "@linkRSPinjectionLinux@\n"
+        ld_rsp = ld_rsp ++ "@linkRSPinjectionLinuxLd@\n"
     else:
         return comp_fail(ctx, "unsupported host for LLVM link metadata: " ++ os() ++ "/" ++ arch())
     if comp_link_path_is_dynamic(libclang):
